askama_parser/src/node.rs

```diff
@@ -635,6 +636,30 @@ pub struct BlockDef<'a> {
 
 impl<'a> BlockDef<'a> {
     fn parse(i: &'a str, s: &State<'_>) -> IResult<&'a str, Self, ParsingError<&'a str>> {
+        fn check_end_block_name<'a>(
+            before: &'a str,
+            after: &'a str,
+            name: &'a str,
+            end_name: &'a str,
+        ) -> IResult<&'a str, &'a str, ParsingError<&'a str>> {
+            if name.is_empty() && !end_name.is_empty() {
+                return Err(nom::Err::Failure(ParsingError {
+                    input: before,
+                    message: Some(Cow::Owned(format!(
+                        "unexpected name `{end_name}` in `endblock` tag for unnamed `block`",
+                    ))),
+                }));
+            } else if !name.is_empty() && !end_name.is_empty() && name != end_name {
+                return Err(nom::Err::Failure(ParsingError {
+                    input: before,
+                    message: Some(Cow::Owned(format!(
+                        "expected name `{name}` in `endblock` tag, found `{end_name}`",
+                    ))),
+                }));
```

Comment:
I think we can deduplicate a bunch of wrapping here?

Reply:
Indeed, good catch!